cmd/eventlistenersink/main.go

```diff
@@ -86,47 +83,44 @@ func main() {
 		logger.Fatal(err)
 	}
 
+	// Create a sharedInformer factory so that we can cache API server calls
 	factory := externalversions.NewSharedInformerFactoryWithOptions(sinkClients.TriggersClient,
 		30*time.Second, externalversions.WithNamespace(sinkArgs.ElNamespace))
 	if sinkArgs.IsMultiNS {
 		factory = externalversions.NewSharedInformerFactory(sinkClients.TriggersClient,
 			30*time.Second)
 	}
 
-	go func(ctx context.Context) {
-		factory.Start(ctx.Done())
-		<-ctx.Done()
-	}(ctx)
-
 	// Create EventListener Sink
 	r := sink.Sink{
-		KubeClientSet:               kubeClient,
-		DiscoveryClient:             sinkClients.DiscoveryClient,
-		DynamicClient:               dynamicCS,
-		TriggersClient:              sinkClients.TriggersClient,
-		HTTPClient:                  http.DefaultClient,
-		EventListenerName:           sinkArgs.ElName,
-		EventListenerNamespace:      sinkArgs.ElNamespace,
-		Logger:                      logger,
-		Auth:                        sink.DefaultAuthOverride{},
+		KubeClientSet:          kubeClient,
+		DiscoveryClient:        sinkClients.DiscoveryClient,
+		DynamicClient:          dynamicCS,
+		TriggersClient:         sinkClients.TriggersClient,
+		HTTPClient:             http.DefaultClient,
+		EventListenerName:      sinkArgs.ElName,
+		EventListenerNamespace: sinkArgs.ElNamespace,
+		Logger:                 logger,
+		Auth:                   sink.DefaultAuthOverride{},
+		// Register all the listers we'll need
 		EventListenerLister:         factory.Triggers().V1alpha1().EventListeners().Lister(),
 		TriggerLister:               factory.Triggers().V1alpha1().Triggers().Lister(),
 		TriggerBindingLister:        factory.Triggers().V1alpha1().TriggerBindings().Lister(),
 		ClusterTriggerBindingLister: factory.Triggers().V1alpha1().ClusterTriggerBindings().Lister(),
 		TriggerTemplateLister:       factory.Triggers().V1alpha1().TriggerTemplates().Lister(),
 		ClusterInterceptorLister:    factory.Triggers().V1alpha1().ClusterInterceptors().Lister(),
 	}
-	eventListenerBackoff := wait.Backoff{
-		Duration: 100 * time.Millisecond,
-		Factor:   2.5,
-		Jitter:   0.3,
-		Steps:    10,
-		Cap:      5 * time.Second,
-	}
-	err = r.WaitForEventListener(eventListenerBackoff)
-	if err != nil {
-		logger.Fatal(err)
+
+	// Start and sync the informers before we start taking traffic
+	// TODO: maybe we should have a timeout here?
```

Comment:
I think something like a 15 second timeout makes sense here

Reply:
In lieu of a timeout, do we have any way to bubble back status to the user to say "we're starting up but we're not healthy / not able to server traffic yet"?

Reply:
Yeah, we don't start listening on the port until the sync is complete. This includes the /live endpoint at the moment (which we use for health checks) --> might have to think through this one a bit more so that we don't end up with endless restart loops if the sync takes too long.

We could try 
1. printing out a log message that we are starting up
2. modify the live endpoint so that we return the above starting up message

Reply:
you could start a goroutine ahead of this that prints out a log line every 5-10s providing some sort of "wait for startup" message. then, cancel the routine after the cache sync has completed in line 123 below.

i think the behavior is correct to not start passing the `/live` check until the informers are fully synced. we can provide default liveness probe settings that will wait for an appropriate time before attempting to restart the pod (or startup probe, but I think we still need to support kube < 1.18).

Reply:
ok, so reading through the code, the sync function will keep going till the stop channel is closed. I'll change the call to pass in a context with a timeout of 1 min which should ensure we don't try forever.

Reply:
Updated and manually tested out that the timeout works.